test/integrations/destinations/klaviyo/processor/screenTestData.ts

```diff
@@ -0,0 +1,98 @@
+import { generatePageOrScreenPayload, transformResultBuilder } from '../../../testUtils';
+
+const destination = {
+  Config: {
+    publicApiKey: 'dummyPublicApiKey',
+    privateApiKey: 'dummyPrivateApiKey',
+  },
+};
+
+export const screenTestData = [
+  {
+    id: 'klaviyo-screen-test-1',
+    name: 'klaviyo',
+    description: 'Screen event call with properties and contextual traits',
+    scenario: 'Business',
+    successCriteria:
+      'Response should contain only event payload and status code should be 200, for the event payload should contain properties and contextual traits in the payload',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            destination,
+            message: generatePageOrScreenPayload(
```

Comment:
utils to generate payload - Is it not an over optimisation & the test case review need to be done at 2 places ?
Is it not better for complete payload to exist in the same file ?

Reply:
From @utsabc 
We can treat these functions as part of test framework, and we should not expect any changes to these functions right.
Secondly the final generated output can be viewed in test reporter for further analysis or review
And finally if total customisation is needed then once can choose to write the payload without functions>
The idea which me and 
[@DKola](https://rudderlabs.slack.com/team/U03JWRHH1NJ)
 discussed was that to reduce the visual complexity of input and output payload for easy review. If complete boiler plate payloads are repeated the input size grows a lot

Reply:
Agree on your points. I just don't want to loose the readability of test case.
Can we avoid adding more parameters to these generate functions.
Instead we can have different functions - generateGroupPayload & generateSimpleGroupPayload